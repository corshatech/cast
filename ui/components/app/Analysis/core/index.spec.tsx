import { render, screen } from '@testing-library/react';
import { AnalysisCard } from './';

const reportedAt = new Date().toISOString();

describe('AnalysisCard', () => {
  it('renders all passed props', () => {
    const { asFragment } = render(<AnalysisCard
      description="Test description"
      weaknessLink="Test weakness link"
      weaknessTitle="Test weakness title"
      title="Test title"
      severity="none"
      reportedAt={reportedAt}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders a weaknessLink without title', () => {
    const { asFragment } = render(<AnalysisCard
      description="Test description"
      weaknessLink="Test weakness link"
      title="Test title"
      severity="none"
      reportedAt={reportedAt}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Does not render a WeaknessLink when given a title but no Link', () => {
    const { asFragment } = render(<AnalysisCard
      description="Test description"
      weaknessTitle="Test weakness title"
      title="Test title"
      severity="none"
      reportedAt={reportedAt}
    />)

    const weaknessTitle = screen.queryByText('Test weakness title')
    // Specifically expect this text to be missing:
    expect(weaknessTitle).toBeNull()
    expect(asFragment()).toMatchSnapshot()
  });
});
