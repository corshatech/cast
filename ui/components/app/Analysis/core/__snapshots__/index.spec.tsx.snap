// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalysisCard Does not render a WeaknessLink when given a title but no Link 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="flex flex-nowrap"
      >
        <h2
          class="leading-6 text-h2 font-poppins font-bold grow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
            data-testid="LightbulbIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19C8.1 2 5 5.1 5 9c0 2.4 1.2 4.5 3 5.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3c1.8-1.3 3-3.4 3-5.7 0-3.9-3.1-7-7-7z"
            />
          </svg>
           Test title
        </h2>
      </div>
      <p
        class="leading-6 text-body1 font-nunito max-w-prose my-2"
      >
        Test description
      </p>
      <table
        class="font-light my-2 text-zinc-400 border-separate border-spacing-x-4"
      >
        <tbody>
          <tr>
            <td>
              Updated:
            </td>
            <td>
              2023-01-17T13:12:00.000Z
            </td>
          </tr>
          <tr>
            <td>
              Issue Severity:
            </td>
            <td>
              none
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AnalysisCard renders a weaknessLink without title 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="flex flex-nowrap"
      >
        <h2
          class="leading-6 text-h2 font-poppins font-bold grow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
            data-testid="LightbulbIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19C8.1 2 5 5.1 5 9c0 2.4 1.2 4.5 3 5.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3c1.8-1.3 3-3.4 3-5.7 0-3.9-3.1-7-7-7z"
            />
          </svg>
           Test title
        </h2>
      </div>
      <p
        class="leading-6 text-body1 font-nunito max-w-prose my-2"
      >
        Test description
      </p>
      <table
        class="font-light my-2 text-zinc-400 border-separate border-spacing-x-4"
      >
        <tbody>
          <tr>
            <td>
              Updated:
            </td>
            <td>
              2023-01-17T13:12:00.000Z
            </td>
          </tr>
          <tr>
            <td>
              Issue Severity:
            </td>
            <td>
              none
            </td>
          </tr>
          <tr>
            <td>
              Learn more:
            </td>
            <td>
              <a
                class="underline"
                href="Test weakness link"
              >
                Test weakness link
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AnalysisCard renders all passed props 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="flex flex-nowrap"
      >
        <h2
          class="leading-6 text-h2 font-poppins font-bold grow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
            data-testid="LightbulbIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19C8.1 2 5 5.1 5 9c0 2.4 1.2 4.5 3 5.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3c1.8-1.3 3-3.4 3-5.7 0-3.9-3.1-7-7-7z"
            />
          </svg>
           Test title
        </h2>
      </div>
      <p
        class="leading-6 text-body1 font-nunito max-w-prose my-2"
      >
        Test description
      </p>
      <table
        class="font-light my-2 text-zinc-400 border-separate border-spacing-x-4"
      >
        <tbody>
          <tr>
            <td>
              Updated:
            </td>
            <td>
              2023-01-17T13:12:00.000Z
            </td>
          </tr>
          <tr>
            <td>
              Issue Severity:
            </td>
            <td>
              none
            </td>
          </tr>
          <tr>
            <td>
              Learn more:
            </td>
            <td>
              <a
                class="underline"
                href="Test weakness link"
              >
                Test weakness title
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AnalysisCardLoading renders as expected 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root animate-pulse css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root space-y-3 css-46bh2p-MuiCardContent-root"
  >
    <div
      className="flex items-center justify-between mb-2"
    >
      <div
        className="flex space-x-2"
      >
        <div
          className="w-8 h-8 bg-gray-200 rounded-full dark:bg-gray-500 "
        />
        <div
          className="h-8 bg-gray-200 rounded-full dark:bg-gray-500 w-64"
        />
      </div>
    </div>
    <span
      className="sr-only"
    >
      Loading...
    </span>
    <div
      className="max-w-prose my-2 h-2.5 bg-gray-200 rounded-full dark:bg-gray-500 w-40 mb-2"
    />
    <div
      className="space-y-2.5 max-w-prose"
    >
      <div
        className="flex items-center w-full space-x-2"
      >
        <div
          className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-32"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full"
        />
      </div>
      <div
        className="flex items-center w-full space-x-2 max-w-[480px]"
      >
        <div
          className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-full"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24"
        />
      </div>
      <div
        className="flex items-center w-full space-x-2 max-w-[400px]"
      >
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full"
        />
        <div
          className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-80"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full"
        />
      </div>
      <div
        className="flex items-center w-full space-x-2 max-w-[480px]"
      >
        <div
          className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-full"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-full"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24"
        />
      </div>
      <div
        className="flex items-center w-full space-x-2 max-w-[440px]"
      >
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-32"
        />
        <div
          className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24"
        />
        <div
          className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-full"
        />
      </div>
    </div>
    <div
      className="w-full bg-gray-200 dark:bg-gray-700 h-[400px] rounded-lg"
    />
  </div>
</div>
`;
